numpy/ma/core.py

```diff
@@ -7683,6 +7676,10 @@ def asanyarray(a, dtype=None):
     <class 'numpy.ma.core.MaskedArray'>
 
     """
+    # workaround for #8666, to preserve identity. Ideally the bottom line
```

Comment:
> One could change `MaskedArray.__new__`

This is the correct approach, and already how `ndarray.__new__` behaves. Now take a look at `MaskedArray.__new__`, and tell me how easy that changes looks. That's why I'm calling this a workaround!